elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/CompletionDsl.scala

```diff
@@ -20,35 +20,57 @@ package com.sumologic.elasticsearch.restlastic.dsl
 
 trait CompletionDsl extends DslCommons with QueryDsl {
 
-  case class Suggest(text: String, completion: Completion) extends Query with RootObject {
+  case class SuggestRoot(textOpt: Option[String], suggestions: List[Suggestion]) extends Query with RootObject {
     val _suggest = "suggest"
     val _text = "text"
+
+
+    override def toJson: Map[String, Any] = {
+      val globalTextJson = if (textOpt.isDefined) Map(_text -> textOpt.get) else Map.empty
+
+      globalTextJson ++ Map(_suggest -> suggestions.map(p => p.name -> p.toJson).toMap)
+    }
+  }
+
+  case class Suggestion(name: String, textOpt: Option[String], termOpt: Option[SuggestionTerm], completionOpt: Option[Completion]) extends Query {
+    val _text = "text"
+    val _term = "term"
     val _completion = "completion"
 
     override def toJson: Map[String, Any] = {
-      Map(_suggest -> Map(
-        _text -> text,
-        _completion -> completion.toJson
-      ))
+      val textJson = if (textOpt.isDefined) Map(_text -> textOpt.get) else Map.empty[String, Any]
```

Comment:
I would to `textOpt.map(text => Map(_text -> text)).getOrElse(Map.empty[String, Any])` for this as well as the lines below

Reply:
Or `Map.empty[String, Any] ++ textOpt.map(text => Map(_text -> text))`